File: common/src/main/java/com/hartwig/actin/algo/datamodel/Evaluation.java
package com.hartwig.actin.algo.datamodel;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class Evaluation {

    @NotNull
    public abstract EvaluationResult result();

    @NotNull
    public abstract Set<String> passMessages();

    @NotNull
    public abstract Set<String> undeterminedMessages();

    @NotNull
    public abstract Set<String> failMessages();
}

File: common/src/main/java/com/hartwig/actin/algo/datamodel/Evaluation.kt
package com.hartwig.actin.algo.datamodel

data class Evaluation(
    val result: EvaluationResult,
    val recoverable: Boolean,
    val inclusionMolecularEvents: Set<String>,
    val exclusionMolecularEvents: Set<String>,
    val passSpecificMessages: Set<String>,
    val passGeneralMessages: Set<String>,
    val warnSpecificMessages: Set<String>,
    val warnGeneralMessages: Set<String>,
    val undeterminedSpecificMessages: Set<String>,
    val undeterminedGeneralMessages: Set<String>,
    val failSpecificMessages: Set<String>,
    val failGeneralMessages: Set<String>,
)
